domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt

```diff
@@ -9,7 +9,7 @@ import org.oppia.util.data.DataProviders
 import org.oppia.util.system.OppiaClock
 import javax.inject.Inject
 
-private const val EXPLORATION_DATA_PROVIDER_ID = "ExplorationDataProvider"
+private const val EXPLORATION_DATA_PROVIDER_ID = "exploration_data_provider_id"
```

Comment:
This does not following the convention mentioned in the issue.
`<function_name>_<optional_context>_PROVIDER_ID`
means that the rename should be 
`GET_EXPLORATION_BY_ID_PROVIDER_ID ="get_exploration_by_id_provider_id"`